site/components/ViewAllContents.tsx

```diff
@@ -0,0 +1,918 @@
+            dateFilterProps.setStartDateFilter("");
+        }
+
+        if (!!dateFilterProps.endDateFilter && success) {
+            if (
+                dateIsSameOrBeforeSecondDate(
+                    getFormattedDate(dateFilterProps.value),
+                    getFormattedDate(dateFilterProps.endDateFilter),
+                )
+            ) {
+                dateFilterProps.setIncompatibleDatesError(false);
+                dateFilterProps.setFilter({
+                    ...dateFilterProps.filter,
+                    period: { startTime: dateFilterProps.value, endTime: dateFilterProps.endDateFilter },
+                });
+            } else {
+                dateFilterProps.setIncompatibleDatesError(true);
+                dateFilterProps.setFilter({ ...dateFilterProps.filter, period: undefined });
+            }
+        } else {
+            dateFilterProps.setFilter({ ...dateFilterProps.filter, period: undefined });
+        }
+    } else {
+        const { success } = dateFilterProps.schema.safeParse(dateFilterProps.value);
+        if (success) {
+            dateFilterProps.setEndDateFilter(dateFilterProps.value);
+        } else {
+            dateFilterProps.setEndDateFilter("");
+        }
+
+        if (!!dateFilterProps.startDateFilter && success) {
+            if (
+                dateIsSameOrBeforeSecondDate(
+                    getFormattedDate(dateFilterProps.startDateFilter),
+                    getFormattedDate(dateFilterProps.value),
+                )
+            ) {
+                dateFilterProps.setIncompatibleDatesError(false);
+                dateFilterProps.setFilter({
+                    ...dateFilterProps.filter,
+                    period: { startTime: dateFilterProps.startDateFilter, endTime: dateFilterProps.value },
+                });
+            } else {
+                dateFilterProps.setIncompatibleDatesError(true);
+                dateFilterProps.setFilter({ ...dateFilterProps.filter, period: undefined });
+            }
+        } else {
+            dateFilterProps.setFilter({ ...dateFilterProps.filter, period: undefined });
+        }
+    }
+};
+
+export const getContentPage = (pageNumber: number, contents: TableContents[]): TableContents[] => {
+    const startPoint = (pageNumber - 1) * 10;
+    const endPoint = pageNumber * 10;
+    return contents.slice(startPoint, endPoint);
+};
+
+export const formatContentsIntoRows = (contents: TableContents[], currentPage: number) => {
+    const contentPages = getContentPage(currentPage, contents);
+    return contentPages.map((content) => {
+        const earliestPeriod: {
+            startTime: string;
+            endTime: string | null;
+        } = content.validityPeriods[0];
+        const latestPeriod: string | null = content.validityPeriods[content.validityPeriods.length - 1].endTime;
+
+        return {
+            header: (
+                <Link
+                    className="govuk-link"
+                    href={
```

Comment:
For these links, we will have to add a check to see if we are in the template user journey and if so add the query param ```?template=true``` to the end of the current pathname string.

Also the return query param should be ```VIEW_ALL_TEMPLATES_PAGE_PATH``` if the user is in the template user journey.